app/reducers/events.js

```diff
@@ -10,227 +10,142 @@ import { normalize } from 'normalizr';
 import { eventSchema } from 'app/reducers';
 import mergeObjects from 'app/utils/mergeObjects';
 import { groupBy, orderBy } from 'lodash';
+import { pull } from 'lodash';
+import produce from 'immer';
 
 export type EventEntity = {
   id: number,
   title: string,
   comments: Array<number>
 };
 
-function mutateEvent(state: any, action: any) {
-  switch (action.type) {
-    case Event.FETCH_PREVIOUS.SUCCESS: {
-      const events = action.payload.result.reduce(
-        (total, id) => ({
-          ...total,
-          [id]: {
-            ...action.payload.entities.events[id],
-            isUsersUpcoming: false
-          }
-        }),
-        {}
-      );
-      return {
-        ...state,
-        byId: mergeObjects(state.byId, events)
-      };
-    }
-    case Event.FETCH_UPCOMING.SUCCESS: {
-      const events = action.payload.result.reduce(
-        (total, id) => ({
-          ...total,
-          [id]: {
-            ...action.payload.entities.events[id],
-            isUsersUpcoming: true
+type State = any;
+
+const mutateEvent = produce(
+  (newState: State, action: any): void => {
+    switch (action.type) {
+      case Event.FETCH_PREVIOUS.SUCCESS:
+        Object.entries(action.payloads.entities.events).forEach(
+          ([eventId, event]) => {
+            newState.byId[eventId] = { ...event, isUsersUpcoming: false };
           }
-        }),
-        {}
-      );
-      return {
-        ...state,
-        byId: mergeObjects(state.byId, events)
-      };
-    }
-    case Event.DELETE.SUCCESS: {
-      return {
-        ...state,
-        items: state.items.filter(id => id !== action.meta.id)
-      };
-    }
-    case Event.SOCKET_EVENT_UPDATED: {
-      const events = normalize(action.payload, eventSchema).entities.events;
-      return {
-        ...state,
-        byId: mergeObjects(state.byId, events)
-      };
-    }
-    case Event.CLEAR: {
-      return {
-        ...state,
-        items: [],
-        pagination: {}
-      };
-    }
-    case Event.REQUEST_REGISTER.BEGIN: {
-      return {
-        ...state,
-        byId: {
-          ...state.byId,
-          [action.meta.id]: {
-            ...state.byId[action.meta.id],
-            loading: true
+        );
+        break;
+
+      case Event.FETCH_UPCOMING.SUCCESS:
+        Object.entries(action.payloads.entities.events).forEach(
```

Comment:
Holy moly :scream:

Reply:
Haha I guess this would be nicer as a simple for loop